### ISSUE TITLE
Failed: akka.http.impl.engine.client.ClientCancellationSpec.(It is not a test it is a sbt.testing.SuiteSelector)

### ISSUE THREAD

**Issue by jrudolph** [akka/akka-http#2954]
https://jenkins.akka.io:8498/job/akka-http-nightly/5/testReport/

```
org.scalatest.exceptions.NotAllowedException: java.net.BindException was thrown inside "Http client connections" must, construction cannot continue: "Cannot assign requested address"
	at org.scalatest.wordspec.AnyWordSpecLike.org$scalatest$wordspec$AnyWordSpecLike$$registerBranch(AnyWordSpecLike.scala:195)
	at org.scalatest.wordspec.AnyWordSpecLike$$anon$1.apply(AnyWordSpecLike.scala:994)
	at org.scalatest.verbs.MustVerb$StringMustWrapperForVerb.must(MustVerb.scala:191)
	at org.scalatest.verbs.MustVerb$StringMustWrapperForVerb.must$(MustVerb.scala:190)
	at org.scalatest.verbs.MustVerb$$anon$1.must(MustVerb.scala:229)
	at akka.http.impl.engine.client.ClientCancellationSpec.<init>(ClientCancellationSpec.scala:20)
```

**Comment by ignasi35** [comment]
Failed to bind a random port (I think there's little to do).

**Comment by jrudolph** [comment]
Yeah, the only way would be to implement #2947 which is needs quite some effort to migrate most tests off. Might still be worthwhile in the long run.

**Comment by jrudolph** [comment]
Fixed in #3010.
